Provider crashes when removing a disk

## Issue
Removing a disk definition from a `proxmox_vm_qemu` resource via commenting causes the provider crash

## Expected Result

The disk is removed from the machine using the ProxMox API.
The following is a definition created for multiple disks on a QEMU virtual machine:

```terraform
  disk {
    slot = var.disks[0].slot
    size = var.disks[0].size
    type = var.disks[0].type
    storage = var.disks[0].storage
    iothread = var.disks[0].iothread
  }

  # disk {
  #   slot = var.disks[1].slot
  #   size = var.disks[1].size
  #   type = var.disks[1].type
  #   storage = var.disks[1].storage
  #   iothread = var.disks[1].iothread
  # }
```
The input for these disks is as follows
```terraform
variable disks {
  description = "Map of disks to configure"

  type=list(
    object(
      {
        slot = number
        size = string
        type = string
        storage = string
        iothread = number
      }
    )
  )

  default = [
    {
      slot = 0
      size = "10G"
      type = "scsi"
      storage = "local-lvm"
      iothread = 1
    }
  ]
}
```

## Result

The provider plugin crashes with the following trace:
```
proxmox_vm_qemu.test_server[0]: Modifying... [id=hostname/qemu/106]
proxmox_vm_qemu.test_server[0]: Still modifying... [id=hostname/qemu/106, 10s elapsed]
proxmox_vm_qemu.test_server[0]: Still modifying... [id=hostname/qemu/106, 20s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   with proxmox_vm_qemu.test_server[0],
│   on main.tf line 30, in resource "proxmox_vm_qemu" "test_server":
│   30: resource "proxmox_vm_qemu" "test_server" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v2.8.0 plugin:

panic: runtime error: index out of range [1] with length 1

goroutine 83 [running]:
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuUpdate(0xc000193280, 0xb7cf00, 0xc0002428c0, 0x0, 0x0)
        github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:1185 +0x39d2
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00019c9a0, 0xda6e58, 0xc0004f4640, 0xc000193280, 0xb7cf00, 0xc0002428c0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:352 +0x1ee
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00019c9a0, 0xda6e58, 0xc0004f4640, 0xc0002441c0, 0xc00007a6c0, 0xb7cf00, 0xc0002428c0, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00012b3b0, 0xda6e58, 0xc0004f4640, 0xc0004aa870, 0xc0004f4640, 0xc768e0, 0xc0002b8c00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000692600, 0xda6f00, 0xc0004f4640, 0xc000244000, 0xc000692600, 0xc0002b8c60, 0xc000112ba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xc768e0, 0xc000692600, 0xda6f00, 0xc0002b8c60, 0xc00007cd20, 0x0, 0xda6f00, 0xc0002b8c60, 0xc0005b0600, 0x10bb)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000268e00, 0xdae5b8, 0xc000138c00, 0xc000300400, 0xc0002cef30, 0x11bbce0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000268e00, 0xdae5b8, 0xc000138c00, 0xc000300400, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000342a0, 0xc000268e00, 0xdae5b8, 0xc000138c00, 0xc000300400)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-proxmox_v2.8.0 plugin crashed!
```

## Metadata
```
Terraform v1.0.8
on linux_amd64
+ provider registry.terraform.io/telmate/proxmox v2.8.0
```

--- a/proxmox/resource_vm_qemu.go
+++ b/proxmox/resource_vm_qemu.go
@@ -1180,8 +1180,16 @@ func resourceVmQemuUpdate(d *schema.ResourceData, meta interface{}) error {
 			// disk added or removed AND there is no disk hot(un)plug
 			d.Set("reboot_required", true)
 		} else {
+			r := len(oldValues)
+
+			// we have have to check if the new configuration has fewer disks
+			// otherwise an index out of range panic occurs if we don't reduce the range
+			if rangeNV := len(newValues); rangeNV < r {
+				r = rangeNV
+			}
+
 			// some of the existing disk parameters have changed
-			for i := range oldValues { // loop through the interfaces
+			for i := 0; i < r; i++ { // loop through the interfaces
 				if oldValues[i].(map[string]interface{})["ssd"] != newValues[i].(map[string]interface{})["ssd"] {
 					d.Set("reboot_required", true)
 				}

